pkg/gameservers/controller.go

```diff
@@ -639,6 +645,52 @@ func (c *Controller) addGameServerHealthCheck(gs *agonesv1.GameServer, pod *core
 	})
 }
 
+func (c *Controller) addSDKServerEnvVars(gs *agonesv1.GameServer, pod *corev1.Pod) {
+	for i, c := range pod.Spec.Containers {
+		if c.Name != sdkserverSidecarName {
+			sdkEnvVars := sdkEnvironmentVariables(gs)
+			if sdkEnvVars == nil {
+				// If a gameserver was created before 1.1 when we started defaulting the grpc and http ports,
+				// don't change the container spec.
+				continue
+			}
+
+			// Filter out environment variables that have reserved names.
+			// From https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating
+			env := c.Env[:0]
+			for _, e := range c.Env {
+				if !reservedEnvironmentVariableName(e.Name) {
+					fmt.Printf("adding %s to final env\n", e.Name)
```

Comment:
Should this be removed? If not, it should probably go through logrus.

Reply:
Yes, left over debugging. As an aside, I couldn't figure out why modifying c in place had no effect and I had to reassign the modified c back to the containers array (hence the debugging).